File: packages/vscode-extension/src/lib/tool-summary.test.ts
import { describe, expect, it } from "vitest";
import {
	buildLiveToolStartPayload,
	summarizeVscodeToolCall,
	withToolSummaryLabels,
} from "./tool-summary.js";

describe("vscode tool summary", () => {
	it("summarizes file tools with leaf paths", () => {
		expect(
			summarizeVscodeToolCall("read", {
				file_path: "/tmp/projects/maestro/package.json",
			}),
		).toBe("Read package.json");
	});

	it("summarizes command tools with the command text", () => {
		expect(
			summarizeVscodeToolCall("exec_command", {
				command: "bun run bun:lint",
			}),
		).toBe("Ran bun run bun:lint");
	});

	it("adds summary labels to message tools", () => {
		const message = withToolSummaryLabels({
			role: "assistant",
			content: "Done",
			timestamp: new Date(0).toISOString(),
			tools: [
				{
					name: "read",
					status: "completed",
					args: { file_path: "/workspace/src/index.ts" },
				},
			],
		});

		expect(message.tools?.[0]?.summaryLabel).toBe("Read index.ts");
	});

	it("preserves raw tool names in live tool payloads", () => {
		expect(
			buildLiveToolStartPayload(
				"read",
				{ path: "/workspace/README.md" },
				{ displayName: "Read File" },
			),
		).toEqual({
			name: "read",
			displayName: "Read File",
			summaryLabel: "Read README.md",
			args: { path: "/workspace/README.md" },
		});
	});
});

File: packages/jetbrains-plugin/src/test/kotlin/com/evalops/composer/toolwindow/ToolSummaryTest.kt
package com.evalops.composer.toolwindow

import com.evalops.composer.api.ComposerToolCall
import com.evalops.composer.api.ToolCallStatus
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ToolSummaryTest {

    @Test
    fun `summarizes read tool calls with leaf path`() {
        val tool = ComposerToolCall(
            name = "read",
            status = ToolCallStatus.COMPLETED,
            args = mapOf("file_path" to "/workspace/src/package.json")
        )

        assertEquals("Read package.json", summarizeToolCall(tool))
    }

    @Test
    fun `summarizes command tools with command text`() {
        val tool = ComposerToolCall(
            name = "exec_command",
            status = ToolCallStatus.RUNNING,
            args = mapOf("command" to "bun run bun:lint")
        )

        assertEquals("Ran bun run bun:lint", summarizeToolCall(tool))
    }

    @Test
    fun `humanizes unknown mcp tool names`() {
        val tool = ComposerToolCall(
            name = "mcp__filesystem__read_file",
            status = ToolCallStatus.COMPLETED,
            args = emptyMap()
        )

        assertEquals("Ran read file", summarizeToolCall(tool))
    }
}
